### PR TITLE
Add action_server implementation

### PR DESCRIPTION
**Pull request by bazfp**
Adds the server half of `action` plugin :)

- Handle Arming / Disarming / Kill
- Handle Flight Mode transitioning

### CODE REVIEW

**Comment by bazfp** [comment]
Requires https://github.com/mavlink/MAVSDK-Proto/pull/244

File: src/plugins/action_server/action_server_impl.cpp [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+void ActionServerImpl::init() {
+        // Arming / Disarm / Kill
+        _parent->register_mavlink_command_handler(
+        MAV_CMD_COMPONENT_ARM_DISARM,
+        [this](const MavlinkCommandReceiver::CommandLong& command) {
+            ActionServer::ArmDisarm armDisarm{
+                command.params.param1 == 1,
+                command.params.param2 == 21196};
+
+            // Check arm states - Ugly.
```
Comment (julianoes) [inline src/plugins/action_server/action_server_impl.cpp:64]: :smile:

File: src/plugins/action_server/action_server_impl.cpp [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+                [this, armDisarm, result]() { _arm_disarm_callback(result, armDisarm); });
+
+            mavlink_message_t msg;
+            mavlink_msg_command_ack_pack(
+                _parent->get_own_system_id(),
+                _parent->get_own_component_id(),
+                &msg,
+                command.command,
+                request_ack,
+                100,
```
Comment (julianoes) [inline src/plugins/action_server/action_server_impl.cpp:95]:
```suggestion
                std::numeric_limits<uint8_t>::max(),
```

Or 255.

Only `RESULT_IN_PROGRESS` shares the progress.

File: src/plugins/action_server/action_server_impl.cpp [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+
+            if (is_custom) {
+                // PX4 uses custom modes
+                px4_custom_mode px4_mode{};
+                px4_mode.main_mode = custom_mode;
+                px4_mode.sub_mode = sub_custom_mode;
+                auto system_flight_mode = _parent->to_flight_mode_from_custom_mode(px4_mode.data);
+                request_flight_mode = telemetry_flight_mode_from_flight_mode(system_flight_mode);
+            } else {
+                // TO DO non PX4 flight modes...
```
Comment (julianoes) [inline src/plugins/action_server/action_server_impl.cpp:123]: Just return `UNSUPPORTED` here?

File: src/plugins/action_server/action_server_impl.cpp [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+            });
+
+            mavlink_message_t msg;
+            mavlink_msg_command_ack_pack(
+                _parent->get_own_system_id(),
+                _parent->get_own_component_id(),
+                &msg,
+                command.command,
+                allow_mode ? MAV_RESULT::MAV_RESULT_ACCEPTED : MAV_RESULT_TEMPORARILY_REJECTED,
+                100,
```
Comment (julianoes) [inline src/plugins/action_server/action_server_impl.cpp:166]: Same here, 255.

File: src/plugins/action_server/include/plugins/action_server/action_server.h [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+     */
+    friend std::ostream& operator<<(std::ostream& str, ActionServer::FlightMode const& flight_mode);
+
+    /**
+     * @brief State to check if the vehicle can transition to
+     * respective flightmodes
+     */
+    struct AllowableFlightModes {
+        bool can_auto_mode{}; /**< @brief Auto/mission mode */
+        bool can_guided_mode{}; /**< @brief Guided mode */
```
Comment (julianoes) [inline src/plugins/action_server/include/plugins/action_server/action_server.h:95]: What do you mean with Guided mode? Guided is basically what we call Offboard in PX4, right?

Comment (bazfp) [reply src/plugins/action_server/include/plugins/action_server/action_server.h:95]: Yes I was basing this on the `base_mode` in mavlink first - it doesn't map to anything at the moment but offboard makes sense.

**Review by julianoes — APPROVED** [review]

**Pull request merged**